lib/isomorphic/create-routes.coffee

```diff
@@ -12,6 +13,15 @@ module.exports = (pages, pagesReq) ->
     handler: pagesReq './_template'
   })
 
+  # Adds a handler for <NotFoundRoute /> that will serve as a 404 Page
+  if config.notFound
+    templates.rootNotFound = Router.createNotFoundRoute({
```

Comment:
This PR is a stopgap as we'll be adding support for "meta pages" soon like * routes through the coming plugin system (https://github.com/gatsbyjs/gatsby/pull/36#issuecomment-142037695). I'll be writing up a RFC soon with some thoughts about how the plugin system will work.